Declarative Web Push: Move the window context PushManager from Navigator to DOMWindow

beidson opened the pull request:
#### b96ecd4d82e2ac0f15580a7a1ea7e54b653d3867
<pre>
Declarative Web Push: Move the window context PushManager from Navigator to DOMWindow
<a href="https://rdar.apple.com/141786288">rdar://141786288</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=284980">https://bugs.webkit.org/show_bug.cgi?id=284980</a>

Reviewed by Chris Dumez.

Since our initial implementation, standards work has moved this from window.navigator directly onto window.

* LayoutTests/platform/mac-wk2/fast/dom/navigator-detached-no-crash-expected.txt:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/push-api/PushStrategy.h:
* Source/WebCore/Modules/push-api/WindowPush.idl: Renamed from Source/WebCore/Modules/push-api/NavigatorPush.idl.
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/page/DOMWindow.cpp:
* Source/WebCore/page/DOMWindow.h:
* Source/WebCore/page/DOMWindow.idl:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::m_pushManager):
(WebCore::LocalDOMWindow::pushManager):
(WebCore::toScope):
(WebCore::LocalDOMWindow::subscribeToPushService):
(WebCore::LocalDOMWindow::unsubscribeFromPushService):
(WebCore::LocalDOMWindow::getPushSubscription):
(WebCore::LocalDOMWindow::getPushPermissionState):
(WebCore::ContextDestructionObserver): Deleted.
* Source/WebCore/page/LocalDOMWindow.h:
* Source/WebCore/page/Navigator.cpp:
(WebCore::Navigator::Navigator):
(WebCore::Navigator::pushManager): Deleted.
(WebCore::toScope): Deleted.
(WebCore::Navigator::subscribeToPushService): Deleted.
(WebCore::Navigator::unsubscribeFromPushService): Deleted.
(WebCore::Navigator::getPushSubscription): Deleted.
(WebCore::Navigator::getPushPermissionState): Deleted.
* Source/WebCore/page/Navigator.h:
* Source/WebCore/page/Navigator.idl:
* Source/WebKit/WebProcess/WebCoreSupport/WebPlatformStrategies.cpp:
(WebKit::WebPlatformStrategies::windowSubscribeToPushService):
(WebKit::WebPlatformStrategies::windowUnsubscribeFromPushService):
(WebKit::WebPlatformStrategies::windowGetPushSubscription):
(WebKit::WebPlatformStrategies::windowGetPushPermissionState):
(WebKit::WebPlatformStrategies::navigatorSubscribeToPushService): Deleted.
(WebKit::WebPlatformStrategies::navigatorUnsubscribeFromPushService): Deleted.
(WebKit::WebPlatformStrategies::navigatorGetPushSubscription): Deleted.
(WebKit::WebPlatformStrategies::navigatorGetPushPermissionState): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebPlatformStrategies.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:
(TestWebKitAPI::subscribe):
(TestWebKitAPI::getPushSubscription):

Canonical link: <a href="https://commits.webkit.org/288144@main">https://commits.webkit.org/288144@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aff5c25ec3471ded41624155d0dbd2e27477d7b7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/82002 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/1529 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/35959 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/86559 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/33034 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/84108 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/1564 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/9353 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/63927 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/33034 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/85072 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/1157 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74613 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/44210 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/1060 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/28790 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/31454 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/72378 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29402 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/87992 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/9244 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/6583 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/72294 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/9429 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70432 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71516 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17827 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15635 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/14551 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/630 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/9195 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/14731 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/9035 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/12560 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/10843 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/page/LocalDOMWindow.h:
```diff
@@ -451,6 +464,17 @@ class LocalDOMWindow final
     RefPtr<CookieStore> m_cookieStore;
+
+#if ENABLE(DECLARATIVE_WEB_PUSH)
+    bool isActive() const final { return true; }
```
I don't think we should mix member functions and data members, even if it is convenient for your ifdefs. Please move the member functions earlier in the header, in their own `#if ENABLE(DECLARATIVE_WEB_PUSH)` block, with other member functions.

cdumez reviewed (approved):
LGTM with fix.

beidson committed:
Declarative Web Push: Move the window context PushManager from Navigator to DOMWindow
rdar://141786288
https://bugs.webkit.org/show_bug.cgi?id=284980

Reviewed by Chris Dumez.

Since our initial implementation, standards work has moved this from window.navigator directly onto window.

* LayoutTests/platform/mac-wk2/fast/dom/navigator-detached-no-crash-expected.txt:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/push-api/PushStrategy.h:
* Source/WebCore/Modules/push-api/WindowPush.idl: Renamed from Source/WebCore/Modules/push-api/NavigatorPush.idl.
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/page/DOMWindow.cpp:
* Source/WebCore/page/DOMWindow.h:
* Source/WebCore/page/DOMWindow.idl:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::m_pushManager):
(WebCore::LocalDOMWindow::pushManager):
(WebCore::toScope):
(WebCore::LocalDOMWindow::subscribeToPushService):
(WebCore::LocalDOMWindow::unsubscribeFromPushService):
(WebCore::LocalDOMWindow::getPushSubscription):
(WebCore::LocalDOMWindow::getPushPermissionState):
(WebCore::ContextDestructionObserver): Deleted.
* Source/WebCore/page/LocalDOMWindow.h:
* Source/WebCore/page/Navigator.cpp:
(WebCore::Navigator::Navigator):
(WebCore::Navigator::pushManager): Deleted.
(WebCore::toScope): Deleted.
(WebCore::Navigator::subscribeToPushService): Deleted.
(WebCore::Navigator::unsubscribeFromPushService): Deleted.
(WebCore::Navigator::getPushSubscription): Deleted.
(WebCore::Navigator::getPushPermissionState): Deleted.
* Source/WebCore/page/Navigator.h:
* Source/WebCore/page/Navigator.idl:
* Source/WebKit/WebProcess/WebCoreSupport/WebPlatformStrategies.cpp:
(WebKit::WebPlatformStrategies::windowSubscribeToPushService):
(WebKit::WebPlatformStrategies::windowUnsubscribeFromPushService):
(WebKit::WebPlatformStrategies::windowGetPushSubscription):
(WebKit::WebPlatformStrategies::windowGetPushPermissionState):
(WebKit::WebPlatformStrategies::navigatorSubscribeToPushService): Deleted.
(WebKit::WebPlatformStrategies::navigatorUnsubscribeFromPushService): Deleted.
(WebKit::WebPlatformStrategies::navigatorGetPushSubscription): Deleted.
(WebKit::WebPlatformStrategies::navigatorGetPushPermissionState): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebPlatformStrategies.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:
(TestWebKitAPI::subscribe):
(TestWebKitAPI::getPushSubscription):

Canonical link: https://commits.webkit.org/288144@main
